library/network/src/lib/y2firewall/firewalld/relations.rb

```diff
@@ -88,7 +88,7 @@ def has_many(*relations, scope: nil, cache: false) # rubocop:disable Style/Predi
         scope = "#{scope}_" if scope
         enable_modifications_cache if cache
 
-        define_method "relations" do
+        define_singleton_method "relations" do
```

Comment:
Well, in case we need it, I would prefer to have both.

```ruby
f = Y2Firewall::Firewalld.instance
external = f.find_zone("external")
external.relations
=> [:services, :interfaces, :protocols, :ports, :sources]
Y2Firewall::Firewalld::Zone.relations
=> [:services, :interfaces, :protocols, :ports, :sources]
```

Reply:
Hm, I was thinking of it from the opposite side. When we have (conceptually)

```
class Zone
  def self.has_many( *relations, ....)
end
```
then it would be nice to query these relations back when analyzing connections between classes. So I've modified that to
```
class Zone
  def self.has_many( *relations, ....)
  def self.relations
end
```
Simple use case is in the pointed yast2-firewall PR (I know that that is not crucial one, but i wanted to learn / play with meta programming a bit ;-)

So, just for curiosity do you have a use case for your case in your mind or do you know / expect any? In my POV it can be easily implemented using delegator - something like ```def_delegator self.class, relations``` What do you think?